service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -148,6 +148,8 @@ else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED
             else {
                 return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.UNPROCESSABLE_ENTITY);
             }
+        } catch (NullPointerException e) {
+            return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR), HttpStatus.SERVICE_UNAVAILABLE);
```

Comment:
Are you relying on the fact that erLookup() right now happens to raise an NPE when the ER returns a non-200 response? That sounds very fragile. Why not raise a specific exception on line 186 instead?

Reply:
It is raising NullPointerException when ER is down and trying to find in an ER. In line 192 of  RemremGenerateController.erLookup(RemremGenerateController.java) it is raising NPE.

Reply:
Yes, but that's an implementation detail that would change if proper null checks are introduced. It's an implementation side-effect, if you will. Since this is a foreseeable and expected error it would be wise to raise a well-defined exception and make it part of the API.